Sources/Regex/Parse/CharacterPropertyClassification.swift

```diff
@@ -0,0 +1,392 @@
+    case "telu", "telugu": return .telugu
+    case "tfng", "tifinagh": return .tifinagh
+    case "tglg", "tagalog": return .tagalog
+    case "thaa", "thaana": return .thaana
+    case "thai": return .thai
+    case "tibt", "tibetan": return .tibetan
+    case "tirh", "tirhuta": return .tirhuta
+    case "tnsa", "tangsa": return .tangsa
+    case "toto": return .toto
+    case "ugar", "ugaritic": return .ugaritic
+    case "vaii", "vai": return .vai
+    case "vith", "vithkuqi": return .vithkuqi
+    case "wara", "warangciti": return .warangCiti
+    case "wcho", "wancho": return .wancho
+    case "xpeo", "oldpersian": return .oldPersian
+    case "xsux", "cuneiform": return .cuneiform
+    case "yezi", "yezidi": return .yezidi
+    case "yiii", "yi": return .yi
+    case "zanb", "zanabazarsquare": return .zanabazarSquare
+    case "zinh", "inherited", "qaai": return .inherited
+    case "zyyy", "common": return .common
+    case "zzzz", "unknown": return .unknown
+    default: return nil
+    }
+  }
+
+  static func classifyCharacterPropertyValueOnly(
+    _ value: String
+  ) throws -> Atom.CharacterProperty.Kind {
+    guard !value.isEmpty else { throw ParseError.emptyProperty }
+
+    // Some special cases defined by UTS#18 (and Oniguruma for 'ANY' and
+    // 'Assigned').
+    switch normalizeForCharacterPropertyMatching(value) {
+    case "any": return .any
+    case "assigned": return .assigned
+    case "ascii": return .ascii
+    default: break
+    }
+
+    // The following properties we can infer keys/values for.
+    if let prop = classifyBoolProperty(value) {
+      return .binary(prop, value: true)
+    }
+    if let cat = classifyGeneralCategory(value) {
+      return .gc(cat)
+    }
+    if let script = classifyScriptProperty(value) {
+      return .script(script)
+    }
+
+    // Some additional special cases we recognise.
+    // TODO: Normalize these?
+    if let oniguruma = OnigurumaSpecialProperty(rawValue: value) {
+      return .onigurumaSpecial(oniguruma)
+    }
+    typealias PCRESpecial = Atom.CharacterProperty.Kind.PCRESpecialCategory
+    if let pcreSpecial = PCRESpecial(rawValue: value) {
+      return .pcreSpecial(pcreSpecial)
+    }
+
+    // Otherwise we don't know what this is.
+    return .other(key: nil, value: value)
+  }
+
+  static func classifyCharacterProperty(
+    key: String, value: String
+  ) throws -> Atom.CharacterProperty.Kind {
+    guard !key.isEmpty && !value.isEmpty else { throw ParseError.emptyProperty }
+
+    if let prop = classifyBoolProperty(key) {
+      let isTrue = try classifyCharacterPropertyBoolValue(value)
+      return .binary(prop, value: isTrue)
+    }
+
+    // This uses the aliases defined in
+    // https://www.unicode.org/Public/UCD/latest/ucd/PropertyAliases.txt.
+    switch normalizeForCharacterPropertyMatching(key) {
+    case "script", "sc":
+      guard let script = classifyScriptProperty(value) else {
+        throw ParseError.invalidScriptKind(value)
+      }
```

Comment:
@milseman Any thoughts on whether we should throw an error here, or just fallback to parsing as an `.other` in case new scripts (and maybe general categories?) get added in the future?

Reply:
Run time can have a different version of the spec which could be newer. I don't know how we'll handle that kind of availability... We may need to communicate back to the parser and our enum/struct or whatever has availability on it.

Reply:
Okay yeah, I've updated to parse unknown cases as `other` for now